### PR TITLE
Update the provider to tf sdk v2

### LINKED ISSUE #7: Update to plugin v2 sdk for terraform
With the release of terraform 1.0, plugin v2 is the recommended way for writing providers. Is it ok if I work on upgrading this to sdk v2?

### PR DESCRIPTION
**Pull request by harsimranmaan**
With the releasen of terraform 1.0, plugin sdk v2 is the recommended version for writing
    provider. v1 has been EoLed.

This PR includes the following changes:
    1. Update plugin sdk version to v2
    2. Update golang to 1.18
    3. Remove vendored golang dependencies
    4. Add Github actions as the CI stage
    5. Remove travis as the CI stage
    6. Run CI on merge requests
    7. Update go-resty to v2
    8. Make the entrollment token a sensitive value

Results for the integration tests are green

```bash
    $ make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v  -timeout 10m
?       github.com/oktadev/terraform-provider-oktaasa   [no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccGroupAssign
--- PASS: TestAccGroupAssign (9.70s)
=== RUN   TestAccGroup
--- PASS: TestAccGroup (3.62s)
=== RUN   TestAccTkn
--- PASS: TestAccTkn (5.27s)
=== RUN   TestAccProject
--- PASS: TestAccProject (7.34s)
PASS
ok      github.com/oktadev/terraform-provider-oktaasa/oktaasa   26.103s
```

Things to follow in subsequent MRs:
    Use context based method for Read, Create, Update and Delete tf methods

Closes #7

### CODE REVIEW

**Comment by harsimranmaan** [comment]
Hi, can someone help review this?

**Comment by harsimranmaan** [comment]
Hi, is anyone able to review and merge this?

File: README.md [hunk 1]
```diff
@@ -10,28 +10,27 @@ Terraform Provider for Okta's Advanced Server Access (Okta's ASA)
...
 
 Building The Provider
 ---------------------
 
-Clone repository to: `$GOPATH/src/github.com/terraform-providers/terraform-provider-oktaasa`
-
+Clone this repository
 ```sh
-$ git clone git@github.com:terraform-providers/terraform-provider-oktaasa $GOPATH/src/github.com/terraform-providers/terraform-provider-oktaasa
+$ git clone git@github.com:terraform-providers/terraform-provider-oktaasa
```
Comment (monde) [inline README.md:21]: I think the clone would now be
```
$ git clone git@github.com:oktadev/terraform-provider-oktaasa
```

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,11 @@
 module github.com/terraform-providers/terraform-provider-oktaasa
```
Comment (monde) [inline go.mod:1]: Aeems like the module should now be `github.com/oktadev/terraform-provider-oktaasa` but that will in turn cause a bunch of imports to change.

**Comment by harsimranmaan** [comment]
Hey @monde @bdemers would appreciate a review, merge and the release of a new version.

**Comment by monde** [comment]
> Hey @monde @bdemers would appreciate a review, merge and the release of a new version.

@harsimranmaan review is on my todo list, hoping to get to it today. Might be tomorrow.

**Pull request closed without merging**